packages/neuron-wallet/src/utils/export-history.ts

```diff
@@ -1,110 +1,269 @@
+    `${headers.map(label => i18n.t(`export-transactions.column.${label}`))}\n`
+  )
+
+  const db = new sqlite3.Database(dbPath, sqlite3.OPEN_READONLY)
+  const dbPromises: any = new Proxy(db, {
+    get(target, key: keyof typeof db, receiver) {
+      if (typeof target[key] === 'function') {
+        return promisify(Reflect.get(target, key, receiver)).bind(target)
+      }
+      return Reflect.get(target, key, receiver)
     }
+  })
+
+  let total: number | undefined
+  let inserted = 0
 
-    let total: number | undefined
-    let inserted = 0
+  const queryInputs = ({ hash, codeHash, hashType }: QueryParams.Inputs): Promise<QueryResponse.Inputs> =>
+    dbPromises.all(
+      `
+      SELECT
+        typeCodeHash,
+        typeArgs,
+        typeHashType,
+        data
+      FROM
+        input
+      WHERE
+        input.transactionHash = ?
+        AND input.typeCodeHash = ?
+        AND input.typeHashType = ?
+        AND input.lockHash IN (${serializedlockHashList})
+      `,
+      [hash, codeHash, hashType]
+    )
 
-    const SEND_TYPE = i18n.t('export-transactions.tx-type.send')
-    const RECEIVE_TYPE = i18n.t('export-transactions.tx-type.receive')
+  const queryOutputs = ({ hash, codeHash, hashType }: QueryParams.Outputs): Promise<QueryResponse.Outputs> => dbPromises.all(
+    `
+    SELECT
+      typeCodeHash,
+      typeArgs,
+      typeHashType,
+      data
+    FROM
+      output
+    WHERE
+      output.transactionHash = ?
+      AND output.typeCodeHash = ?
+      AND output.typeHashType = ?
+      AND output.lockHash IN (${serializedlockHashList})
+    `,
+    [hash, codeHash, hashType]
+  )
 
-    const writeStream = fs.createWriteStream(filePath)
-    writeStream.write(
-      `${['time', 'block-number', 'tx-hash', 'tx-type', 'amount', 'description']
-        .map(label => i18n.t(`export-transactions.column.${label}`))}\n`
+  const aggrQuery = ({ hash }: QueryParams.Aggr): Promise<QueryResponse.Aggr> =>
+    dbPromises.get(
+      `
+      SELECT
+        CAST(SUM(CAST(output.capacity AS UNSIGNED BIG ING)) AS VARCHAR) outputShannon,
+        COUNT(output.daoData) daoCellCount
+      FROM
+        output
+      WHERE
+        output.transactionHash = ?
+        AND output.lockHash IN (${serializedlockHashList})
+      GROUP BY
+        output.transactionHash
+      `,
+      [hash]
     )
 
-    const db = new sqlite3.Database(dbPath, sqlite3.OPEN_READONLY)
-    const serializedlockHashList = lockHashList.map(l => `'${l}'`).join(`,`)
-
-    const onRowLoad = (err: Error, row: {
-      hash: string,
-      inputShannon: string | null,
-      timestamp: string,
-      blockNumber: string,
-      description: string
-    }) => {
-      db.serialize(() => {
+  const tokenInfoList: { tokenID: string, symbol: string | null, decimal: string | null }[] = await dbPromises.all(
+    `
+    SELECT
+      tokenID,
+      symbol,
+      decimal
+    FROM
+      sudt_token_info
+    `
+  )
+
+  return new Promise((resolve, reject) => {
```

Comment:
I would suggest encapsulating these logics under a return operator into a function as it is quite long. Encapsulating in a function makes this code more readable.

Reply:
I'll commit a refactor in this PR.

Reply:
The refactor is committed: https://github.com/nervosnetwork/neuron/pull/1723/commits/1479d87e17bb18dbd8f3264a1e2bd4ea10a17353